Mathlib/Analysis/NormedSpace/LogarithmicDerivatives.lean

```diff
@@ -0,0 +1,132 @@
+/-
+Copyright (c) 2024 Chris Birkbeck. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chris Birkbeck
+-/
+import Mathlib.Analysis.Complex.LocallyUniformLimit
+import Mathlib.Analysis.SpecialFunctions.Trigonometric.Deriv
+import Mathlib.Analysis.SpecialFunctions.Complex.LogDeriv
```

Comment:
These two imports are used only for the examples; I might be inclined to put those into a separate file, probably also under `Analysis.SpecialFunctions`. This opinion is weakly held for now.

Reply:
They don't seem important enough for their own file, but I could put them at the start of the file proving the cotangent identity once its in?

Reply:
We should have a file about `logDeriv` in `Analysis/Calculus`, then we can import this file in our `SpecialFunctions` files (old ones or new ones depending on imports) and compute log derivatives of these functions there.

Reply:
Sorry, I don't quite understand. Do you mean make a file about `logDeriv` in `Analysis/Calculus`?

Reply:
Yes, please move basic facts that are true in any nontrivially normed field to `Analysis/Calculus/LogDeriv`.

Reply:
Ok so I did the following: I made a new file `Analysis/Calculus/LogDeriv` and put everything that was true for nontrivially normed fields (which was most of the stuff in the file). I then made a file in `SpecialFunctions/Trigonometric` with the log derivs of the trig functions. I left the rest in the `LogarithmicDerivatives` file which I've renamed to `LogDeriv` for consistency. 

Hopefully this is what you meant? But let me know if not.

We also already have a file `Analysis/SpecialFunctions/Complex/LogDeriv` which could maybe contain the link between LodDerivs and the derivative of composing with log, but I've not modified it for the moment as I'm not sure if we want to add more imports to this file.